File: features/digital/common_digital/src/main/java/com/tokopedia/common_digital/cart/domain/IDigitalCartRepository.java
package com.tokopedia.common_digital.cart.domain;

import com.tokopedia.common_digital.cart.data.entity.requestbody.atc.RequestBodyAtcDigital;
import com.tokopedia.common_digital.cart.data.entity.requestbody.checkout.RequestBodyCheckout;
import com.tokopedia.common_digital.cart.view.model.cart.CartDigitalInfoData;
import com.tokopedia.common_digital.cart.view.model.checkout.InstantCheckoutData;

import rx.Observable;

/**
 * Created by Rizky on 27/08/18.
 */
public interface IDigitalCartRepository {

    Observable<CartDigitalInfoData> addToCart(
            RequestBodyAtcDigital requestBodyAtcDigital, String idemPotencyKeyHeader
    );

    Observable<InstantCheckoutData> instantCheckout(RequestBodyCheckout requestBodyCheckout);

}

File: features/recharge/common_digital/src/main/java/com/tokopedia/common_digital/cart/domain/IDigitalCartRepository.kt
package com.tokopedia.common_digital.cart.domain

import com.tokopedia.common_digital.cart.data.entity.requestbody.checkout.RequestBodyCheckout
import com.tokopedia.common_digital.cart.view.model.checkout.InstantCheckoutData
import rx.Observable

/**
 * Created by Rizky on 27/08/18.
 */
interface IDigitalCartRepository {

    fun instantCheckout(requestBodyCheckout: RequestBodyCheckout): Observable<InstantCheckoutData>

}